tools/buildhelper/buildhelper.go

```diff
@@ -26,39 +27,117 @@ import (
 	"github.com/mdomke/git-semver/version"
 )
 
-func showVersion() int {
+const (
+	develBranchName     = "devel"
+	releaseBranchPrefix = "release-"
+)
+
+type buildInfo struct {
+	Branch  string `json:"branch"`
+	Version string `json:"version"`
+	Commit  string `json:"commit"`
+}
+
+func getVersion() (string, error) {
 	if ver, ok := os.LookupEnv("NRO_BUILD_VERSION"); ok {
-		fmt.Println(ver)
-		return 0
+		return ver, nil
 	}
 
 	v, err := version.Derive()
+	return v.String(), err
+}
+
+func getCommit() (string, error) {
+	if cm, ok := os.LookupEnv("NRO_BUILD_COMMIT"); ok {
+		return cm, nil
+	}
+
+	cmd := exec.Command("git", "log", "-1", "--pretty=format:%h")
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(out)), nil
+}
+
+func getBranch() (string, error) {
+	if cm, ok := os.LookupEnv("NRO_BUILD_BRANCH"); ok {
```

Comment:
where are these env vars set? for such data I don't see the use case where we need to allow overriding the real fetched values or I'm missing a detail